xarray/conventions/netcdftimeindex.py

```diff
@@ -106,9 +120,23 @@ def get_date_type(self):
     return type(self._data[0])
 
 
+def assert_all_same_netcdftime_datetimes(data):
+    from netcdftime._netcdftime import datetime
+
+    if not isinstance(data[0], datetime):
+        raise TypeError(
+            'NetCDFTimeIndex requires netcdftime._netcdftime.datetime'
```

Comment:
Use the public API name `netcdftime.datetime`.

Also, print the invalid object in the error message (using `.format`)